ios/SecureStorage/SecureStorage.swift

```diff
@@ -0,0 +1,105 @@
+    if status == errSecItemNotFound { return nil }
+    let statusMessage = SecCopyErrorMessageString(status, nil) as? String
+    guard status == errSecSuccess else {
+      throw SecureStorageError(description: statusMessage ?? "", code: Int(status))
+    }
+    return result as? Data
+  }
+  
+  
+  /// Removes a previously stored data.
+  /// - Parameter key: the key of the data to be removed.
+  /// - Throws: An error of type `SecureStorageError` if the operation fails which wraps the message and the error code.
+  public func remove(
+    key: String
+  ) throws {
+    let query: [String: Any] = [kSecClass: kSecClassGenericPassword, kSecAttrService: serviceName, kSecAttrAccount: key] as [String: Any]
+    let status = SecItemDelete(query as CFDictionary);
+    let statusMessage = SecCopyErrorMessageString(status, nil) as? String
+    guard status == errSecSuccess || status == errSecItemNotFound else { //Ignore the errSecItemNotFound when calling remove with an invalid key
+      throw SecureStorageError(description: statusMessage ?? "", code: Int(status))
+    }
+  }
+  
+  
+  /// Clears any value previsously stored.
+  /// - Throws: An error of type `SecureStorageError` if the operation fails which wraps the message and the error code.
+  public func clear() throws {
+    let query: [String: Any] = [kSecClass: kSecClassGenericPassword, kSecAttrService: serviceName] as [String: Any]
+    let status = SecItemDelete(query as CFDictionary);
+    let statusMessage = SecCopyErrorMessageString(status, nil) as? String
+    guard status == errSecSuccess || status == errSecItemNotFound else { //Ignore the errSecItemNotFound when calling clear on an empty keychain
+      throw SecureStorageError(description: statusMessage ?? "", code: Int(status))
+    }
+  }
+  
+  /// Returns an array of the keys in the secure storage.
+  /// - Returns: an array of strings containing the keys currently used to store data. An empty array if there isn't any key.
+  /// - Throws: An error of type `SecureStorageError` if the operation fails which wraps the message and the error code.
+  public func keys() throws -> [String] {
+    let query: [String: Any] = [kSecClass: kSecClassGenericPassword, kSecAttrService: serviceName, kSecReturnRef: true, kSecReturnData: true, kSecReturnAttributes: true, kSecMatchLimit: kSecMatchLimitAll] as [String: Any]
+    var result: AnyObject?
+    let status = SecItemCopyMatching(query as CFDictionary, &result)
+    if status == errSecItemNotFound { return [] }
+    let statusMessage = SecCopyErrorMessageString(status, nil) as? String
+    guard status == errSecSuccess else {
+      throw SecureStorageError(description: statusMessage ?? "", code: Int(status))
+    }
+    
+    return (result as? [[String: Any]])?.compactMap {
+      guard let account = $0[kSecAttrAccount as String] as? String else {
+        return nil // Skip if kSecAttrAccount is not a String
+      }
+      return account
+    } ?? []
```

Comment:
Better for readability? What do you think?

```suggestion
    if let resultArray = result as? [[String: Any]] {
        return resultArray.compactMap { item in
            guard let account = item[kSecAttrAccount as String] as? String else {
                return nil // Skip if kSecAttrAccount is not a String
            }
            return account
        }
    }
    return []
```

Reply:
Way better, thanks